src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -292,6 +298,65 @@ public void mergeStagingLoadHistoryTable(Dataset dataset, String flightId)
     bigQueryProject.query(sqlTemplate.render());
   }
 
+  private static final String getLoadHistoryTemplate =
+      "SELECT * "
+          + "FROM `<project>.<dataset>.<loadTable>` L "
+          + "WHERE L.load_tag = @loadTag "
+          + "LIMIT <limit> "
+          + "OFFSET <offset>";
+
+  public List<BulkLoadHistoryModel> getLoadHistory(
+      Dataset dataset, String loadTag, int offset, int limit) {
+    BigQueryProject bigQueryProject = BigQueryProject.from(dataset);
+    String datasetName = prefixName(dataset.getName());
+    var sqlTemplate = new ST(getLoadHistoryTemplate);
+    sqlTemplate.add("project", bigQueryProject.getProjectId());
+    sqlTemplate.add("dataset", datasetName);
+    sqlTemplate.add("loadTable", PDAO_LOAD_HISTORY_TABLE);
+    sqlTemplate.add("limit", limit);
+    sqlTemplate.add("offset", offset);
+
+    var query = sqlTemplate.render();
+    QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(query)
+            .addNamedParameter("loadTag", QueryParameterValue.string(loadTag))
+            .build();
+
+    try {
+      var results = bigQueryProject.getBigQuery().query(queryConfig);
+      return StreamSupport.stream(results.iterateAll().spliterator(), true)
+          .map(BigQueryPdao::bigQueryResultToBulkLoadHistoryModel)
+          .collect(Collectors.toList());
+    } catch (InterruptedException ex) {
+      throw new GoogleResourceException("Could not query BigQuery for load history", ex);
+    }
+  }
+
+  private static BulkLoadHistoryModel bigQueryResultToBulkLoadHistoryModel(
+      FieldValueList fieldValue) {
+    Function<FieldValue, String> emptyToNull =
+        fv -> {
+          var value = fv.getStringValue();
+          return value.isEmpty() ? null : value;
+        };
+    var model =
+        new BulkLoadHistoryModel()
+            .sourcePath(fieldValue.get(LoadHistoryUtil.SOURCE_NAME_FIELD_NAME).getStringValue())
+            .targetPath(fieldValue.get(LoadHistoryUtil.TARGET_PATH_FIELD_NAME).getStringValue())
+            .state(
+                BulkLoadFileState.fromValue(
+                    fieldValue.get(LoadHistoryUtil.STATE_FIELD_NAME).getStringValue()))
+            .fileId(fieldValue.get(LoadHistoryUtil.FILE_ID_FIELD_NAME).getStringValue());
+
+    Optional.ofNullable(fieldValue.get(LoadHistoryUtil.CHECKSUM_CRC32C_FIELD_NAME))
+        .ifPresent(o -> model.checksumCRC(emptyToNull.apply(o)));
+    Optional.ofNullable(fieldValue.get(LoadHistoryUtil.CHECKSUM_MD5_FIELD_NAME))
+        .ifPresent(o -> model.checksumMD5(emptyToNull.apply(o)));
+    Optional.ofNullable(fieldValue.get(LoadHistoryUtil.ERROR_FIELD_NAME))
+        .ifPresent(o -> model.error(emptyToNull.apply(o)));
```

Comment:
I think I've already said this in this PR (or maybe another one like it?) but since `null` is the valid state for these fields when no value is set you could instead write:
```
    var model =
        new BulkLoadHistoryModel()
            .sourcePath(fieldValue.get(LoadHistoryUtil.SOURCE_NAME_FIELD_NAME).getStringValue())
            [...]
            .checksumCRC(emptyToNull.apply(fieldValue.get(LoadHistoryUtil.CHECKSUM_CRC32C_FIELD_NAME)))
```